api/openai.py

```diff
@@ -46,95 +45,89 @@ def __init__(self):
-                loop = asyncio.get_running_loop()
-                loop.create_task(utils.log_error(f"Possible issue with the OpenAI API. Status: {http_response.status_code}, Content: {http_response.text}"))
+                self.log_error(
+                    f"Possible issue with the OpenAI API. Status: {http_response.status_code}, Content: {http_response.text}"
+                )
                 return True
             response = http_response.json()
+
         else:
             try:
                 response = Moderation.create(input=text)
             except openai.error.AuthenticationError as e:
-                self.log.error(self.class_name, error="OpenAI Authentication Failed")
-                loop = asyncio.get_running_loop()
-                loop.create_task(utils.log_error(f"OpenAI Authenication Failed"))
-                loop.create_task(utils.log_exception(e))
+                self.log_error("OpenAI Authentication Failed", e)
                 return True
             except openai.error.RateLimitError as e:
-                self.log.warning(self.class_name, error="OpenAI Rate Limit Exceeded")
-                loop = asyncio.get_running_loop()
-                loop.create_task(utils.log_error(f"OpenAI Rate Limit Exceeded"))
-                loop.create_task(utils.log_exception(e))
+                self.log_error(self.class_name, "OpenAI Rate Limit Exceeded", e, warning=True)
                 return True
 
-        flagged: bool = response["results"][0]["flagged"]
-
-        all_morals: frozenset[str] = ["sexual", "hate", "harassment", "self-harm", "sexual/minors", "hate/threatening", "violence/graphic", "self-harm/intent", "self-harm/instructions", "harassment/threatening", "violence"]
-        violated_categories = set()
+        results = response.get("results", [])[0]
+        if not results:
+            return False
 
-        if flagged:
-            for moral in all_morals - allowed_categories:
-                if response["results"][0][moral]:
-                    violated_categories.add(moral)
+        if not results["flagged"]:
+            self.log.info(self.class_name, msg=f"Checked with content filter, it says the text looks clean")
+            return False
 
-        if len(violated_categories) > 0:
+        violated_categories = [
+            moral for moral in OPENAI_NASTY_CATEGORIES - allowed_categories if results.get(moral)
+        ]
+        if violated_categories:
             self.log.warning(self.class_name, msg=f"Text violated these unwanted categories: {violated_categories}")
             self.log.debug(self.class_name, msg=f"OpenAI moderation response: {json.dumps(response)}")
             return True
-        else:
-            self.log.info(self.class_name, msg=f"Checked with content filter, it says the text looks clean")
-            return False
+
+        self.log.info(self.class_name, msg=f"Checked with content filter, it says the text looks clean")
```

Comment:
@mruwnik can we update the message to `"Checked with content filter, it doesn't violate any of our categories"`, please?

because it was flagged (not "clean"), just not with any category we care about

Reply:
done